File: 7-kyu/method-for-counting-total-occurence-of-specific-digits/python/solution.py
class List(object):
    def count_spec_digits(self, integers_list, digits_list):
        ret = []
        int_str_list = [list(str(i)) for i in integers_list]
        dig_str_list = [str(dig) for dig in digits_list]
        for dig_str in dig_str_list:
            cnt = (int(dig_str), 0)
            for integer in int_str_list:
                cnt = (cnt[0], cnt[1] + integer.count(dig_str))
            ret.append(cnt)
        return ret

File: 7-kyu/method-for-counting-total-occurence-of-specific-digits/javascript/solution.js
function List() {
  this.countSpecDigits = function(integersList, digitsList) {
    const countDigits = (n, digit) => n.toString().split("").filter(elem => parseInt(elem) === digit).length;
    let ret = [];
    for (const digit of digitsList) {
      let count = 0;
      for (const integer of integersList) {
        count += countDigits(integer, digit);
      }
      ret.push([digit, count]);
    }
    return ret;
  };
}
